core/state_transition.go

```diff
@@ -222,7 +222,10 @@ func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bo
 		}
 	}
 	st.refundGas()
-	st.state.AddBalance(st.evm.Coinbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.gasPrice))
+	fee := new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.gasPrice)
+	if !st.evm.ChainConfig().IsDarvaza(st.evm.BlockNumber) {
+		st.state.AddBalance(st.evm.Coinbase, fee)
```

Comment:
Just out of curiosity, what is this doing with the coinbase address?

Reply:
This is just the second half of the gas fee being transferred - it has already been charged and refunded before here. Now it's conditional to be pre-darvaza only.